### DISCUSSION TITLE
Logging

### DISCUSSION THREAD

**Discussion by Abhishek2271** [Trusted-AI/adversarial-robustness-toolbox#1362] [Q&A] [answered]
Hi guys,

I have been using ART to generate adversarial examples using multiple algorithms and so far it works really smoothly. I have some problems with logging. I see that there are logging instances in the evasion attack files but ART does not create log files.
So for instance UAP.py has
logger.info("Success rate of universal perturbation attack: %.2f%%", 100 * fooling_rate)
but this does not log anything and no log files are created.
Shouldn't log files be created in the same directory as program execution?

**Comment by beat-buesser** [comment, marked as answer]
Hi @Abhishek2271 Thank you very much for using ART! Are you already defining a logger before running ART? This example show how to set up the default logger that will be used by ART: https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/examples/adversarial_training_cifar10.py
That way you can set the logging level to capture more information and also define a file to save the logs, etc.

**Reply by Abhishek2271** [reply]
Hi @beat-buesser ,

I had been using the Tensorpack default logger and it seems that it somehow does not work. I created a new instance of logger as you suggested and it works. For my debugging purposes, this is enough. Thank you for your reply.
